src/coreclr/vm/methodtable.cpp

```diff
@@ -7765,11 +7752,23 @@ Module *MethodTable::GetDefiningModuleForOpenType()
     RETURN NULL;
 }
 
+PCODE MethodTable::GetRestoredSlotIfExists(DWORD slotNumber)
```

Comment:
Is the "If exists" concept handled in the `GetSlot()`? Looking at this implementation the name is very unsettling.

Reply:
Based on my reading of the existing code the only difference here is an assert. We should not be creating extra functions with guarantees that are enforced only by asserts.